transport_plugins/bled112/iotile_transport_bled112/bled112_auth.py

```diff
@@ -164,12 +164,15 @@ def authenticate(self, uuid, client_supported_auth, handshake_fn: SendAuthHandsh
         Perform authentication
 
         Args:
-            uuid (int or str): either device mac or id
+            uuid (int): the device UUID
             auth_type (int): See .AuthType
             handshake_fn (SendAuthHandshakeFn): a function pointer
         Returns:
             (bool, bytearray/dict): tuple of success flag, session key or dict with reason of failure
         """
+        if not uuid or not isinstance(uuid, int):
```

Comment:
this should be `uuid is None` instead of `not uuid`.   `not uuid` returns True if `uuid == 0`, which is a potentially valid device_id you shouldn't reject.

Reply:
Good catch. Fixed